### PR TITLE
Add blog feature

### PR DESCRIPTION
**Pull request by Alex-ond**

### CODE REVIEW

File: src/app/blog/store/blog.effects.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import * as BlogActions from "./blog.actions";
+import { catchError, map, of, switchMap } from "rxjs";
+import { PostsService } from "../services/posts.service";
+
+@Injectable()
+export class BlogEffects {
+
+    constructor(private actions$: Actions, private postsService: PostsService) {}      
+
+    fetchPosts$ = createEffect(() =>
```
Comment (viktor-gusev) [inline src/app/blog/store/blog.effects.ts:12]: If you open comments and get back /posts request is sent again.

You may prevent it by doing something like:
```ts
  fetchPosts$ = createEffect(() =>
    this.actions$.pipe(
      ofType(BlogActions.fetchPosts),
      withLatestFrom(this.store.select(postsSelector)),
      switchMap(([action, storedPosts]) => {
        if (storedPosts.length > 0) return of(BlogActions.fetchPostsSuccess({posts: storedPosts}))
        return this.postsService.fetchPosts().pipe(
          map(posts => BlogActions.fetchPostsSuccess({posts})),
          catchError((error: Error) => of(BlogActions.fetchPostsFailure({errorMessage: error.message}))
          ));
      })));

Comment (viktor-gusev) [reply src/app/blog/store/blog.effects.ts:12]: And in reducer:
```ts
    on(BlogActions.fetchPosts, (state) => {
        return { 
            ...state, 
            isPostsLoading: true,
        };
    }),

File: src/app/blog/blog/post/post-comments/post-comments.component.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    this.paramsSubscription = this.route.params
+      .subscribe(
+        (params: Params) => {
+          const postId = +params['id'];
+          this.store.dispatch(fetchPostCommentsByPostId(postId));
+        }
+      );
+  }
+
+  ngOnDestroy() {
```
Comment (viktor-gusev) [inline src/app/blog/blog/post/post-comments/post-comments.component.ts:34]: There is this feature in late Angular versions:
https://v17.angular.io/api/core/rxjs-interop/takeUntilDestroyed
```ts
this.route.params
      .pipe(takeUntilDestroyed())
      .subscribe(

**Review by viktor-gusev — APPROVED** [review]
lgtm

File: src/app/shared/pipes/orderBy.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+    name: 'orderBy',
+    pure: false
+  })
+  export class OrderByPipe implements PipeTransform {  
+    transform(entities: any[], direction: 'asc' | 'desc', propName: string): any[] {
```
[the remark this answers is not in the document]
Comment (Alex-ond) [reply src/app/shared/pipes/orderBy.pipe.ts]: Changed to 
  transform<O extends object, K extends keyof O>(
        entities: O[] | null, 
        filterString: string, 
        propertyName: K): O[] {

**Pull request merged**